src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -314,18 +324,18 @@ private Task OnEventMessageAsync(ServiceEventMessage message)
 
         private Task OnAccessKeyMessageAsync(AccessKeyResponseMessage keyMessage)
         {
-            if (string.IsNullOrEmpty(keyMessage.ErrorType))
+            if (HubEndpoint.AccessKey is AadAccessKey key)
             {
-                if (HubEndpoint.AccessKey is AadAccessKey key)
+                if (string.IsNullOrEmpty(keyMessage.ErrorType))
                 {
                     key.UpdateAccessKey(keyMessage.Kid, keyMessage.AccessKey);
                 }
-            }
-            else
-            {
-                var exception = new AzureSignalRUnauthorizedException(new AzureSignalRException(keyMessage.ErrorMessage));
-                Log.AuthorizeFailed(Logger, exception);
-                return Task.CompletedTask;
+                else if (key.HasExpired)
+                {
+                    var exception = new AzureSignalRUnauthorizedException(new AzureSignalRException(keyMessage.ErrorMessage));
```

Comment:
we don't need to create an exception only for logging purposes. How about improving Log to log the exception message?